### PR TITLE
[Fix #1926] Do not crash when correcting a hash with a splat in it

### LINKED ISSUE #1926: rubocop v0.29.1 crashed when auto-correct hash with double splat (**)
Given this file:

```ruby
Hash(foo: 'bar',
  **extra_params
)
```

when you run: `bundle exec rubocop -DR --auto-correct`, will yield errors:

```
$ rcc null/example.rb
Inspecting 1 file

0 files inspected, no offenses detected
undefined method `loc' for nil:NilClass
/Users/Juan/.gem/ruby/2.2.2/gems/rubocop-0.29.1/lib/rubocop/cop/style/align_hash.rb:239:in `block in autocorrect'
/Users/Juan/.gem/ruby/2.2.2/gems/rubocop-0.29.1/lib/rubocop/cop/corrector.rb:46:in `call'
/Users/Juan/.gem/ruby/2.2.2/gems/rubocop-0.29.1/lib/rubocop/cop/corrector.rb:46:in `block in rewrite'
/Users/Juan/.gem/ruby/2.2.2/gems/rubocop-0.29.1/lib/rubocop/cop/corrector.rb:45:in `each'
/Users/Juan/.gem/ruby/2.2.2/gems/rubocop-0.29.1/lib/rubocop/cop/corrector.rb:45:in `rewrite'
/Users/Juan/.gem/ruby/2.2.2/gems/rubocop-0.29.1/lib/rubocop/cop/team.rb:83:in `autocorrect_one_cop'
/Users/Juan/.gem/ruby/2.2.2/gems/rubocop-0.29.1/lib/rubocop/cop/team.rb:65:in `autocorrect'
/Users/Juan/.gem/ruby/2.2.2/gems/rubocop-0.29.1/lib/rubocop/cop/team.rb:35:in `inspect_file'
/Users/Juan/.gem/ruby/2.2.2/gems/rubocop-0.29.1/lib/rubocop/runner.rb:129:in `inspect_file'
/Users/Juan/.gem/ruby/2.2.2/gems/rubocop-0.29.1/lib/rubocop/runner.rb:100:in `block in do_inspection_loop'
/Users/Juan/.gem/ruby/2.2.2/gems/rubocop-0.29.1/lib/rubocop/runner.rb:94:in `loop'
/Users/Juan/.gem/ruby/2.2.2/gems/rubocop-0.29.1/lib/rubocop/runner.rb:94:in `do_inspection_loop'
/Users/Juan/.gem/ruby/2.2.2/gems/rubocop-0.29.1/lib/rubocop/runner.rb:72:in `process_file'
/Users/Juan/.gem/ruby/2.2.2/gems/rubocop-0.29.1/lib/rubocop/runner.rb:53:in `block in inspect_files'
/Users/Juan/.gem/ruby/2.2.2/gems/rubocop-0.29.1/lib/rubocop/runner.rb:51:in `each'
/Users/Juan/.gem/ruby/2.2.2/gems/rubocop-0.29.1/lib/rubocop/runner.rb:51:in `inspect_files'
/Users/Juan/.gem/ruby/2.2.2/gems/rubocop-0.29.1/lib/rubocop/runner.rb:30:in `run'
/Users/Juan/.gem/ruby/2.2.2/gems/rubocop-0.29.1/lib/rubocop/cli.rb:24:in `run'
/Users/Juan/.gem/ruby/2.2.2/gems/rubocop-0.29.1/bin/rubocop:13:in `block in <top (required)>'
/Users/Juan/.rubies/ruby-2.2.2/lib/ruby/2.2.0/benchmark.rb:303:in `realtime'
/Users/Juan/.gem/ruby/2.2.2/gems/rubocop-0.29.1/bin/rubocop:12:in `<top (required)>'
/Users/Juan/.gem/ruby/2.2.2/bin/rubocop:23:in `load'
/Users/Juan/.gem/ruby/2.2.2/bin/rubocop:23:in `<main>'
```

And for some reason, I cannot upgrade my rubocop (locked at v0.29.1) :sweat: 

Any pointers on how could I fix this? Thanks!

### PR DESCRIPTION
**Pull request by rrosenblum**
This fixes #1926.

### CODE REVIEW

**Comment by jonas054** [comment]
I think you should also add `**extra_params` in some (or maybe all) of the existing autocorrect examples, just to see that we support this operator for different configured styles and in explicit hashes (with braces). There's also the problem of Ruby 1.9 not having the same support for the double splat operator in hash literals as in Ruby 2.

**Comment by rrosenblum** [comment]
> I think you should also add **extra_params in some (or maybe all) of the existing autocorrect examples

Good idea, I will add more double splat examples

> There's also the problem of Ruby 1.9 not having the same support for the double splat operator in hash literals as in Ruby 2.

I did not realize that Ruby 1.9 did not support the double splat. That explains why I was having trouble reproducing/fixing this in Ruby 1.9.

File: spec/rubocop/cop/style/align_hash_spec.rb [hunk 1]
```diff
@@ -212,6 +212,29 @@
       expect(new_sources).to eq(['hash = { a: 1, b: 2,',
                                  '         c: 3 }'].join("\n"))
     end
+
+    context 'ruby >= 2.0', ruby_greater_than_or_equal: 2.0 do
+      it 'auto-corrects alignment when using double splat ' \
+         'in a hash constructor' do
```
Comment (bbatsov) [inline spec/rubocop/cop/style/align_hash_spec.rb]: Technically speaking, this is not a hash constructor.

**Comment by rrosenblum** [comment]
I fixed the wording in the test

**Comment by bbatsov** [comment]
:+1:

**Pull request merged**